File: leetcode/new/leetcode/boats-to-save-people.php
class Solution {

    /**
     * @param Integer[] $people
     * @param Integer $limit
     * @return Integer
     */
    function numRescueBoats($people, $limit) {
        sort($people);
        $left = 0;
        $right = sizeof($people) - 1;
        $out = 0;
        while ($right>= $left){
            if($people[$left] + $people[$right] <= $limit){
                $left++;
            }
            $out++;
            $right--;
        }
        return $out;
    }
}

File: leetcode/boats-to-save-people.py
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        left = out = 0
        right = len(people) - 1
        while right >= left:
            if people[left] + people[right] <= limit:
                left += 1
            out += 1
            right -= 1
        
        return out